.github/workflows/test.yml

```diff
@@ -80,3 +80,72 @@ jobs:
       # can't use systemd driver with cgroupv1
       if: matrix.rootless != 'rootless'
       run: sudo -E PATH="$PATH" script -e -c /bin/bash -c 'stty rows 40 cols 80; TERM=xterm make RUNC_USE_SYSTEMD=yes local${{ matrix.rootless }}integration'
+
+
+  # cgroup v2 unified hierarchy + very recent kernel (openat2)
+  fedora:
+    # nested virtualization is only available on macOS hosts
+    runs-on: macos-10.15
+    timeout-minutes: 60
+    # only run it if others have passed
+    needs: [test]
+    steps:
+      - uses: actions/checkout@v2
+      - name: prepare vagrant
+        run: |
+          ln -sf Vagrantfile.fedora33 Vagrantfile
+          vagrant up
+          vagrant ssh-config >> ~/.ssh/config
+
+      - name: system info
+        run: ssh default 'sh -exc "uname -a && systemctl --version && df -T"'
+
+      - name: unit tests
+        run: ssh default 'cd /vagrant && sudo make localunittest'
+
+      # The integration tests require tty which GH actions lack;
+      # wrap those in "script" to emulate tty.
+      - name: cgroupv2 with systemd
+        run: ssh default "script -e -c /bin/bash -c 'stty rows 40 cols 80; TERM=xterm sudo make -C /vagrant localintegration RUNC_USE_SYSTEMD=yes'"
```

Comment:
maybe `script` and `ssty` stuff should be moved to Makefile, but can be worked out later

Reply:
Yes, this looks kinda ugly. I thought about a shell script or a program that checks if tty is available, and re-executes itself through script (and does other needed things) if not, and then execute whatever is passed on in arguments.

GHA is the only environment without a tty, so it does not make sense to have this in Makefile (at least unconditionally).

Let's do it later.